### DISCUSSION TITLE
An embeddable playground for Ruby using ruby.wasm

### DISCUSSION THREAD

**Discussion by hatemhosny** [ruby/ruby.wasm#389] [Show and tell]
Great project. Thank you!

Please allow me to introduce [LiveCodes](https://livecodes.io), a [feature-rich](https://livecodes.io/docs/features/), [open-source](https://github.com/live-codes/livecodes), [client-side](https://livecodes.io/docs/why/#client-side), code playground that supports [80+ languages and frameworks](https://livecodes.io/docs/why/#language-support). The [announcement post](https://blog.livecodes.io/introducing-livecodes/) gives a general overview.

**Ruby** has a first-class support in LiveCodes using ruby.wasm (and also Opal).

Ruby (wasm) starter template is available for a quick start:  
https://livecodes.io/?template=ruby-wasm  
(see all [starter templates](https://livecodes.io/?new)).

![Ruby (wasm) starter template in LiveCodes](https://github.com/ruby/ruby.wasm/assets/10176547/232060c2-ad0c-4aa2-a65f-e3700d93e5c5)

You can start an empty Ruby (wasm) project using this link: https://livecodes.io/?ruby-wasm

Projects can be [shared](https://livecodes.io/docs/features/share/), [exported](https://livecodes.io/docs/features/export/) and [deployed](https://livecodes.io/docs/features/deploy/) (to GitHub Pages).

Playgrounds can be [embedded](https://livecodes.io/docs/features/embeds) in any webpage, using a powerful, yet easy-to-use, [SDK](https://livecodes.io/docs/sdk).

This is an example for an embedded Ruby (wasm) playground:

```html
<script type="module">
  import { createPlayground } from "https://unpkg.com/livecodes";

  createPlayground("#container", {
    config: {
      activeEditor: "script",
      markup: { 
        language: "html",
        content: "<h1></h1>",
      },
      script: {
        language: "ruby-wasm",
        content: 'require "js"\ndocument = JS.global[:document]\ndocument.querySelector("h1")[:innerHTML] = "Hello, Ruby!"\n',
      },
    },
  });
</script>
```
[Preview](https://livecodes.io/?x=code/N4IgLglmA2CmIC4QFUB2kawCYAIAKATgPYBWsAxmCADQhawDO5BEADpEaoiDSABawAhlm4AeALawwgnOT6CCDKQF4AOigAqAMQC0ADnU4A9AD5VqCVJmpBktSABuEWAHdWRAmEPlOYWOnsXCCwwPmV6J3JYHSCQvmocCFQoCEFoHSY02GUARgA6AAZDU14+MHFoAEEwMEVuaEFUAHN7f28GhgZ7dV5pJoZEAG0AXVpBSggHWABRLCgPbiYWdl5xBQBrAFdWRFAG5s3BJvgkMoreH3R-KiRRPhyTUSN7kxAAX1oGMABPOF2QfZNQ7HbjkToXXzXbjvT7MNg3PaNIFHE4gAibABG3xiggY4ghV3Q3AIsAAjpsICScOoSAx1OYsERyJtJOgcMocAApADKeSa0CIGLSgwQjOZrLAwwZTJZ1zy5NgBG+3NgcEoHgAFOp7uoAJQipKoRUACQ0AFkADLDdnUkDG1UChIAJUx3wAhPSuB8QF9fowBFIBggRrDlmAgyGQGCGIRGFJobRWMQop0PBHRlHNl8iOIVTUkv1dt6IOJ3J4g8BvT9WIwi7Q-F8KwCkcDUdXGHCVrRLn4iUgYY5FQwIJxuAAmMe8EnCTjQb6IABmaSUY2gApcFqRAGF5M0TrVNrBaOIiPRuAvNmvekQiNAm-5BBi4CIkGloLxxpApqDOAxb-BPmkMAs1BAUlBEb0AC8b3xBAcjeIA)

Screenshot:

![embedded Ruby (wasm) playground](https://github.com/ruby/ruby.wasm/assets/10176547/fd516ab5-0304-4331-b81f-c1581bff1fad)

Please refer to the [SDK documentations](https://livecodes.io/docs/sdk) for details.

Comprehensive [documentations](https://livecodes.io/docs/) are available with live demos, code samples and screenshots.

LiveCodes is free with unlimited usage, no ads and no account required. It can be easily [self-hosted](https://livecodes.io/docs/features/self-hosting) (if you want), and can be used for commercial projects ([MIT license](https://livecodes.io/docs/license)).

Disclosure: obviously, I'm the author of LiveCodes.

I thought you would be interested. 
Thank you.

**Comment by Largo** [comment]
Looks interesting! Do you manually update the ruby.wasm version?

**Reply by hatemhosny** [reply]
Thank you 

Yes, manually set. Currently using [`ruby-3_2-wasm-wasi@2.1.0`](https://www.npmjs.com/package/ruby-3_2-wasm-wasi/v/2.1.0)

**Reply by hatemhosny** [reply]
Check LiveCodes docs for ruby.wasm here: https://livecodes.io/docs/languages/ruby-wasm
